.github/workflows/ci-build.yml

```diff
@@ -1,59 +1,289 @@
+          pip install .[docs]
           make -C doc html
           touch doc/build/html/.nojekyll
 
       - name: Upload Documentation
-        uses: actions/upload-artifact@v2.2.1
+        uses: actions/upload-artifact@v3
         with:
           name: Documentation
           path: doc/build/html
           retention-days: 7
 
       - name: Deploy Docs
-        uses: JamesIves/github-pages-deploy-action@4.1.0
+        uses: JamesIves/github-pages-deploy-action@v4.4.0
         if: startsWith(github.event.ref, 'refs/tags')
         with:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           BRANCH: gh-pages
           FOLDER: doc/build/html
           CLEAN: true
 
-      - name: Lint package dist
+  package:
+    name: Package library
+    runs-on: ubuntu-latest
+    needs: [docs-build, core-import]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+
+      - name: Cache pip
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: Python-${{ runner.os }}-${{ env.MAIN_PYTHON_VERSION }}-${{ hashFiles('pyproject.toml') }}-core
+          restore-keys: |
+            Python-${{ runner.os }}-${{ env.MAIN_PYTHON_VERSION }}
+
+      - name: Install dependencies and build the library
         run: |
-          pip install twine
-          python setup.py sdist
-          twine check dist/*
+          python -m pip install --upgrade pip poetry twine
+          python -m poetry build
+          python -m twine check dist/*
 
-      - name: Upload to PyPi
-        if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+      - name: Install PyAnsys from wheel
         run: |
-          twine upload --skip-existing dist/*
+          pip install dist/*.whl
+
+      - name: Upload wheel and binaries
+        uses: actions/upload-artifact@v3
+        with:
+          name: PyAnsys-packages
+          path: dist/
+          retention-days: 7
+
+  release:
+    name: Release project
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
+    needs: [docs-build, core-import, package]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.MAIN_PYTHON_VERSION }}
+
+      - uses: actions/download-artifact@v3
+
+      - name: Display structure of downloaded files
+        run: ls -R
+
+      - name: Upload to Public PyPi
+        run: |
+          pip install twine
+          twine upload --skip-existing ./**/*.whl
+          twine upload --skip-existing ./**/*.tar.gz
         env:
-          TWINE_USERNAME: "__token__"
-          TWINE_PASSWORD: ${{ secrets.TWINE_TOKEN }}
-          TWINE_REPOSITORY_URL: "https://upload.pypi.org/legacy/"
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
```

Comment:
Do we want a dedicated token for the metapackage?
I can provide one.

Reply:
That sounds right. There is already a PyPi project for this https://pypi.org/project/pyansys/ Could you get access to it?